projects/ngrx.io/content/guide/signals/signal-store/testing.md

```diff
@@ -0,0 +1,687 @@
+    expect(store.loading()).toBe(false);
+  });
+});
+```
+
+By making use of the testing framework's function to manage time, we can verify both scenarios.
+
+This test also uses a setup function to avoid code duplication. This is a common pattern in testing and an alternative to the `beforeEach` function. In our case, every test can decide if it wants to use the setup function or not.
+
+#### with Signals
+
+Testing both scenarios with a type of `Signal` as an input is similar to Observables.
+
+This is mainly due because of the asynchronous tasks we use here.
+
+```typescript
+describe('MoviesStore', () => {
+  // ... beforeEach, afterEach, and setup omitted
+
+  it('should test two sequential loads with a Signal', async () => {
+    const store = setup();
+    const studio = signal('Warner Bros');
+    store.load(studio);
+
+    await jest.advanceTimersByTimeAsync(100);
+    expect(store.movies()).toEqual([{ id: 1, name: 'Harry Potter' }]);
+
+    studio.set('Universal');
+    await jest.advanceTimersByTimeAsync(100);
+    expect(store.movies()).toEqual([{ id: 2, name: 'Jurassic Park' }]);
+  });
+
+  it('should cancel a running request when a new one is made via a Signal', async () => {
+    const store = setup();
+    const studio = signal('Warner Bros');
+
+    effect(() => {
+      console.log(studio());
+    });
+    store.load(studio);
+
+    await jest.advanceTimersByTimeAsync(50);
+
+    studio.set('Universal');
+    await jest.advanceTimersByTimeAsync(50);
+    expect(store.movies()).toEqual([]);
+    expect(store.loading()).toBe(true);
+
+    await jest.advanceTimersByTimeAsync(50);
+    expect(store.movies()).toEqual([{ id: 2, name: 'Jurassic Park' }]);
+    expect(store.loading()).toBe(false);
+  });
+});
+```
+
+Be aware of the glitch-free effect when using Signals. The `rxMethod` relies on `effect`, which might need to be triggered manually via `TestBed.flushEffects()`.
+
+If the mocked `MovieService` operates synchronously, the following test would fail without calling `TestBed.flushEffects()`.
+
+```typescript
+describe('MoviesStore', () => {
+  // ... beforeEach, and afterEach omitted
+
+  it('should depend on flushEffects because of synchronous execution', () => {
+    const moviesService = {
+      load: jest.fn((studio: string) => of([studio === 'Warner Bros' ? { id: 1, name: 'Harry Potter' } : { id: 2, name: 'Jurassic Park' }])),
+    };
+
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: MoviesService,
+          useValue: moviesService,
+        },
+      ],
+    });
+
+    const store = TestBed.inject(MoviesStore);
+    const studio = signal('Warner Bros');
+    store.load(studio);
+    TestBed.flushEffects(); // required
+    expect(store.movies()).toEqual([{ id: 1, name: 'Harry Potter' }]);
+
+    studio.set('Universal');
+    TestBed.flushEffects(); // required
+    expect(store.movies()).toEqual([{ id: 2, name: 'Jurassic Park' }]);
+  });
+});
+```
+
+### Custom extensions
```

Comment:
A thought: would this be better if it would be a recipe?
This page is lengthy (which isn't a bad thing), and I think the main focus should be on testing a signal store.

Reply:
We can also add the "Testing" section to the "Custom Store Features" page, but I'd keep it here to have the full testing guide of SignalStore and related APIs in a single place.

Reply:
We don't have or require special Testing APIs, that need documentation. The SignalStore is an Angular service and service testing is covered by the official Angular docs.
As Tim said, it is, therefore, more a recipe page than a testing guide. 

I see three main sections:
1. Testing the SignalStore
2. Mocking the SignalStore
3. Integration Tests

I would see "Testing Custom Store Features" as a subchapter of "Testing the SignalStore".